pkg/apis/cartographer/v1alpha1/workload_helpers.go

```diff
@@ -200,6 +202,24 @@ func (w *WorkloadSpec) MergeParams(key string, value interface{}) {
 	w.Params = append(w.Params, param)
 }
 
+func (w *WorkloadSpec) MergeComplexParams(key string, value interface{}) {
+	var obj interface{}
+	switch value.(type) {
+	case string:
+		s := value.(string)
+		s = strings.ReplaceAll(s, `\n`, "\n")
+		s = strings.ReplaceAll(s, `\t`, "\t")
+		err := yaml.Unmarshal([]byte(s), &obj)
+		if err != nil {
+			fmt.Println(fmt.Errorf("The provided value for param-yaml %q is not valid yaml/json: %s", key, err))
+			os.Exit(1)
```

Comment:
Can you research about adding unit test for invalid input? this is critical path and it would be good to have test coverage

idea: https://stackoverflow.com/questions/26225513/how-to-test-os-exit-scenarios-in-go

Reply:
Changed `os.Exit`, used validation error instead